### PR TITLE
This kills the deep fried foods holder. Refactors deep frying to just make the thing edible but still functional.

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Refactors deepfrying, removing the gross Deep Fried Foods Holder Object and replacing it with the edible component. 

Now, deep frying a food will simply make the item edible directly. This means it's still functional and doesn't become a dead item. 

This follows the same method that grilling uses when applying its effects. Tweaks grilling a bit so they line up better. Also, silver foods can make grilled items. 

image

I swear this is unrelated to the other 2 fried foods related PRs. I started this a few weeks ago.

## Why It's Good For The Game

Tangibly better code (doesn't have to copy a million vars! Less abusable!) at the price of removing a soulful piece of code. Also means that deep frying an item doesn't irreversibly make it unusable / dead. This is sad, but... damn the holder object sucks. 

Unfortunate side effect is that anything that overrides `attack` to not send signal will *not* be edible when deepfried. Maybe this encourages better signal use? 

Either that or fried foods can override `pre_attack` to hook directly into eating. I can do that as well. 

## Changelog

:cl: Melbert
refactor: Refactored deep fried foods. Deep fried foods are still ""usable"" as their normal item, but are just edible.
qol: Silver Slime stuff can spawn grilled as well as fried. 
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
wow, I was just about to ask why we hadn't done this yet

deep fried disabler

:salute:

**Comment by Jacquerel** [comment]
This may be the end to deep fried Ian, alas.

File: code/datums/elements/food/fried_item.dm [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/// Items fried through the deep fryer.
+/datum/element/fried_item
+	/// List of colors to apply to fried food, each index is applied at a different level
+	var/static/list/fried_colors = list(
+		rgb(166, 103, 54),
```
Comment (Mothblocks) [inline code/datums/elements/food/fried_item.dm:5]: These should be color defines, it's impossible for me to know what these are on web without looking them up

File: code/datums/elements/food/fried_item.dm [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+			this_food.desc = "A heavily-fried... something. Who can tell anymore?"
+
+	ADD_TRAIT(this_food, TRAIT_FOOD_FRIED, "kfc")
+	// Already edible items will inherent these parameters
+	// Otherwise, we will become edible.
+	this_food.AddComponent(/datum/component/edible, \
+		bite_consumption = 2, \
+		food_flags = FOOD_FINGER_FOOD, \
+		junkiness = 10, \
+		foodtypes = FRIED)
```
Comment (Mothblocks) [inline code/datums/elements/food/fried_item.dm:46]:
```suggestion
	this_food.AddComponent( \
		/datum/component/edible, \
		bite_consumption = 2, \
		food_flags = FOOD_FINGER_FOOD, \
		junkiness = 10, \
		foodtypes = FRIED,
	)
```

File: code/datums/elements/food/fried_item.dm [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+			this_food.add_atom_colour(fried_colors[3], FIXED_COLOUR_PRIORITY)
+			this_food.name = "deep-fried [this_food.name]"
+			this_food.desc += " Deep-fried to perfection."
+
+		if(85 to INFINITY)
+			this_food.add_atom_colour(fried_colors[4], FIXED_COLOUR_PRIORITY)
+			this_food.name = "\proper the physical manifestation of the very concept of fried foods"
+			this_food.desc = "A heavily-fried... something. Who can tell anymore?"
+
+	ADD_TRAIT(this_food, TRAIT_FOOD_FRIED, "kfc")
```
Comment (Mothblocks) [inline code/datums/elements/food/fried_item.dm:39]: Make "kfc" a define or just use `REF(src)` instead

File: code/datums/elements/food/fried_item.dm [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+		junkiness = 10, \
+		foodtypes = FRIED)
+
+/datum/element/fried_item/Detach(atom/source, ...)
+	for(var/color in fried_colors)
+		source.remove_atom_colour(FIXED_COLOUR_PRIORITY, color)
+	source.name = initial(source.name)
+	source.desc = initial(source.desc)
+	REMOVE_TRAIT(source, TRAIT_FOOD_FRIED, "kfc")
+	// We will not bother making it un-edible again, it will get too messy
```
Comment (Mothblocks) [inline code/datums/elements/food/fried_item.dm:54]: 👀❓ I don't think it's that bad to just qdel the edible component even if you don't yet verify where it's from

Comment (Mothblocks) [reply code/datums/elements/food/fried_item.dm:54]: You can also save a list of weakrefs of edible components added by the element and check if the `GetComponent` result is one of those before qdel'ing. That will require adding `ELEMENT_DETACH_ON_HOST_DESTROY` though.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/datums/elements/food/fried_item.dm (hunk 4 above)
Comment (Mothblocks) [reply code/datums/elements/food/fried_item.dm]: Right

File: code/datums/elements/food/fried_item.dm [hunk 5]
```diff
@@ -1,11 +1,12 @@
...
 /datum/element/fried_item
-	/// List of colors to apply to fried food, each index is applied at a different level
+	/// List of colors to apply the element target.
+	/// Each index corresponds to a different level.
 	var/static/list/fried_colors = list(
-		rgb(166, 103, 54),
-		rgb(103, 63, 24),
-		rgb(63, 23, 4),
-		rgb(33, 19, 9),
+		"#996666",
```
Comment (Mothblocks) [inline code/datums/elements/food/fried_item.dm:6]: Color defines, aka the ones in colors.dm

Comment (Mothblocks) [reply code/datums/elements/food/fried_item.dm:6]: We probably have some of these already or at least close to them

**Comment by TemporalOroboros** [comment]
A few notes:
1. Most of the things on the oilfry blacklist are there because of the frying rendering them unusable, such as containers. If that's being changed it can probably be trimmed a bit.
2. This will kill the primary method of creating recursive food. If you want to preserve that you're going to want to take a look at custom reagent holders.
3.

File: code/datums/elements/food/fried_item.dm (hunk 5 above)
Comment (Mothblocks) [reply code/datums/elements/food/fried_item.dm:6]: You should add the brown defines then

**Comment by Colovorat** [comment]
An end of an Era

**Comment by moocowswag** [comment]
is this a balance pr or was using deep frying to disable people`s items an exploit.

File: code/__DEFINES/dcs/signals/signals_food.dm [hunk 6]
```diff
@@ -13,12 +13,6 @@
 /// called when an edible ingredient is added: (datum/component/edible/ingredient)
 #define COMSIG_FOOD_INGREDIENT_ADDED "edible_ingredient_added"
 
-// Deep frying foods
-/// From obj/item/food/deepfryholder/Initialize
-#define COMSIG_ITEM_FRIED "item_fried"
```
Comment (Mothblocks) [inline code/__DEFINES/dcs/signals/signals_food.dm:18]: Can you keep the signal, even if you have no consumers? It is the best way to implement #71550.

**Comment by private-tristan** [comment]
Thank you melbert now a clown with cooking oil and 3 seconds in the armory no longer destroys every single gun 👍

**Comment by ATH1909** [comment]
> It unironically is a big gameplay change for jani borgs since their main asimov tool is filling their drying agent sprayer with frying oil and spraying weapons after slipping people mid lethal combat to prevent human harm.

you have a trash bag, my dude

**Comment by Overuseddart** [comment]
> is this a balance pr or was using deep frying to disable people`s items an unintended exploit.
> 
> It unironically is a big gameplay change for jani borgs since their main asimov tool is filling their drying agent sprayer with frying oil and spraying weapons after slipping people mid lethal combat to prevent human harm.
> 
> Maybe a way to wash fried items to restore them could also be on the table cause the permanent change to sprite + permanently edible is kind of oof? Maybe like with soap or something so you can make even normally edible food inedible and dont have to worry about checking for if the item was originally a non edible or not.

You could always just run away with it while they are trying to get up

**Comment by Mothblocks** [comment]
did you figure it out yesterday too

**Comment by Mothblocks** [comment]
i think the feature removal is unfortunate but i also dont really mind

**Pull request merged**